eval/eval_api.go

```diff
@@ -170,7 +171,8 @@ func AddEvalResult(name, code string) error {
 func EvalString(this any, code string, emptyEnv bool) (object.Object, error) {
 	l := lexer.New(code)
 	p := parser.New(l)
-	program := p.ParseProgram()
+	var program ast.Node
+	program = p.ParseProgram()
```

Comment:
I don't get this change

Reply:
ParseProgram returns *ast.Statements which happens to be a node (one of the subtype) but when we call Modify() later it returns a Node and we re-assign it (modified copy) into same variable so it has to be of the right type